relayer/crates/starknet-integration-tests/src/tests/ics20.rs

```diff
@@ -409,22 +410,31 @@ fn test_starknet_ics20_contract() -> Result<(), Error> {
             balance_cosmos_a_step_1.quantity + transfer_quantity
         );
 
-        // TODO(rano): we should use the poseidon hash of the ibc denom to get the token address
+        let ics20_token_address: Felt = {
+            let ibc_prefixed_denom = PrefixedDenom {
+                trace_path: vec![TracePrefix {
+                    port_id: ics20_port.to_string(),
+                    channel_id: starknet_channel_id.channel_id.clone(),
+                }],
+                base: Denom::Hosted(denom_cosmos.to_string()),
+            };
+
+            // https://github.com/informalsystems/ibc-starknet/blob/e64a8ecaa708c5c5150b058b6c9bbe1ba9f54d51/cairo-contracts/packages/utils/src/utils.cairo#L36
+            let ibc_prefixed_denom_key = PoseidonState::digest(&[PoseidonState::digest(
+                &cairo_encoding.encode(&ibc_prefixed_denom)?,
+            )]);
```

Comment:
The `.key()` method in Cairo uses,

https://github.com/informalsystems/ibc-starknet/blob/e64a8ecaa708c5c5150b058b6c9bbe1ba9f54d51/cairo-contracts/packages/utils/src/utils.cairo#L36

I am not sure why the double hashing is necessary. Why can't we use the following?

```rs
poseidon_hash_span(self.data.span())
```